src/main/java/org/opensearch/security/filter/SecurityFilter.java

```diff
@@ -305,7 +305,7 @@ public int order() {
             final PrivilegesEvaluatorResponse pres = eval.evaluate(user, action, request, task, injectedRoles);
             
             if (log.isDebugEnabled()) {
-                log.debug(pres);
+                log.debug(pres.toString());
```

Comment:
This is creating a log message, which you probably don't want. It could be interpolated.

The preferred equivalent way to do this is:

```
log.debug("{}", pres);
```

But, I recommend making the log line even more readable:

```
log.debug("privileges response: {}", pres);
```

Reply:
Here is a reference: https://www.slf4j.org/faq.html#string_contents